File: algebra/derivative_of_polynomial/java/derivative.java
import java.util.Scanner;

public class DerivativePolynomial {
 
 
public static void main(String[] args) {
  
  Scanner sc = new Scanner(System.in);
  
  System.out.print("Enter the degree : ");
  
  int degree = sc.nextInt();
  
  System.out.println("Enter the " + (degree+1) + " coefficients " );
  
  int[] coeff = new int[degree+1];
  
  for (int i = 0; i < coeff.length; i++) {
   coeff[i] = sc.nextInt();
  }
  
  int[] derivativeCoeff = new int[degree] ;  
  
  for (int i = 0; i < derivativeCoeff.length;  i++) {
   derivativeCoeff[i] = coeff[i]*(degree - i);
  }
  
  for (int i = 0; i < derivativeCoeff.length; i++) {
   System.out.print(derivativeCoeff[i] + "*x^" + (degree - i) );
   if(i != derivativeCoeff.length-1 && derivativeCoeff[i+1] > 0  ) {
    System.out.print("+");
   }
  }
  
 }

}

File: algebra/derivative_of_polynomial/python/derivative.py
#!/usr/bin/env python


def polynomial_derivative(degree, coefficients):
    for i in range(degree):
        coefficients[i] *= i #Multiply coefficient by index
    coefficients.pop(0)
    return degree - 1, coefficients

def polynomial_show(degree, coefficients):
    string_representation = ""
    for i in range(degree):
        coefficient = coefficients[i]
        
        if i != 0:
            if coefficients[i] >= 0:
                string_representation += " + "
            else:
                string_representation += " - "
                coefficient = abs(coefficient)
        
        string_representation += f"{coefficient}x^{i}"
    return string_representation

if __name__ == '__main__':
    degree = int(input("Enter degree of polynomial (highest power):"))+1
    coefficients = []
    for i in range(degree):
        coefficient = int(input(f"Enter the coefficient at the x^{i} term:"))
        coefficients.append(coefficient)
    
    print("Original:")
    print(polynomial_show(degree, coefficients))
    
    degree, coefficients = polynomial_derivative(degree, coefficients)
    
    print("Derivative:")
    print(polynomial_show(degree, coefficients))
    
    
